static/js/util.js

```diff
@@ -290,26 +290,35 @@ export function clean_user_content_links(html) {
             elt.removeAttribute("target");
         }
 
-        // Ensure that the title displays the real URL.
-        let title;
-        let legacy_title;
-        if (url.origin === window.location.origin && url.pathname.startsWith("/user_uploads/")) {
-            // We add the word "download" to make clear what will
-            // happen when clicking the file.  This is particularly
-            // important in the desktop app, where hovering a URL does
-            // not display the URL like it does in the web app.
-            title = legacy_title = $t(
-                {defaultMessage: "Download {filename}"},
-                {filename: url.pathname.slice(url.pathname.lastIndexOf("/") + 1)},
+        // We do want to preserve the original title for inline images because they
+        // open in a lightbox instead of being downloaded.
+        const is_inline_image =
+            elt.parentElement && elt.parentElement.classList.contains("message_inline_image");
+        if (!is_inline_image) {
```

Comment:
What's the thinking about this conditional? I think essentially all of our images have `message_inline_image`, so this seems like it'd result in the below code being a noop.

Reply:
OK, I think that makes sense; we should perhaps improve the comments to make this more obvious.